src/instructlab/model/chat.py

```diff
@@ -667,16 +668,23 @@ def start_prompt(
             self.info["messages"].pop()
             raise ChatException("Rate limit exceeded") from e
         except openai.APIConnectionError as e:
+            logger.debug("Connection error, try again...", exc_info=True)
             self.console.print("Connection error, try again...", style="red bold")
             self.info["messages"].pop()
             raise KeyboardInterrupt from e
         except KeyboardInterrupt as e:
             raise e
         except httpx.RemoteProtocolError as e:
+            logger.debug("Connection to the server was closed", exc_info=True)
             self.console.print("Connection to the server was closed", style="bold red")
             self.info["messages"].pop()
             raise ChatException("Connection to the server was closed") from e
         except Exception as e:
+            if logger.getEffectiveLevel() == logging.DEBUG:
```

Comment:
Suggestion: Would it be better to use the if to form a different message but still raise in the exception in the same place?  I would worry the code paths might accidentally diverge and change behavior (not just messages) when debug is enabled.